markdownlint-cli2-config-schema.json

```diff
@@ -0,0 +1,83 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"title": "markdownlint-cli2 configuration schema",
+	"type": "object",
+	"properties": {
+		"config": {
+			"$ref": "./node_modules/markdownlint/schema/markdownlint-config-schema.json"
+		},
+		"customRules": {
+			"description": "Module names or paths of custom rules to load and use when linting",
+			"type": "array",
+			"items": {
+				"description": "Module name or path of a custom rule",
+				"type": "string"
+			}
+		},
+		"fix": {
+			"description": "Whether to enable fixing of linting errors reported by rules that emit fix information",
+			"type": "boolean",
+			"default": false
+		},
+		"frontMatter": {
+			"description": "Regular expression used to match and ignore any front matter at the beginning of a document",
+			"type": "string"
+		},
+		"ignores": {
+			"description": "Glob expressions to ignore when linting",
+			"type": "array",
+			"items": {
+				"description": "Glob expression to ignore",
+				"type": "string",
+				"minLength": 1
+			}
+		},
+		"markdownItPlugins": {
+			"description": "markdown-it plugins to load and use when linting",
+			"type": "array",
+			"items": {
+				"description": "Name or path of a markdown-it plugin followed by parameters",
+				"type": "array",
+				"items": [
+					{
+						"description": "Name or path of a markdown-it plugin",
+						"type": "string"
+					},
+					{
+						"description": "Parameter to pass to the markdown-it plugin"
+					}
```

Comment:
Is it exhaustive to say type here is string|object|boolean|array? I'd prefer to be explicit. But if that's too cumbersome, this works.

Reply:
Also, this literally seems to say there should be 2 items in the array? Is there a way to express that this is optional and repeating?

Reply:
As far as I can tell, omitting the `type` will accept all valid JSON types, which would be desired in order to pass arbitrary parameter values to object formatters and `markdown-it` plugins. 

That being said, I don't mind being explicit. Added the following in b1d7282:

```json
["string", "integer", "number", "boolean", "null", "array", "object"]
```

Regarding the `"minItems: 1"` that I'm using for `markdownItPlugins` and `outputFormatters`: while reading the JSON schema spec I noticed that you can specify multiple types for each item of an array. The format of these config properties are:

```json
"markdownItPugins": [
  ["some-plugin"],
  ["some-other-plugin", "param 0", { "more": "params" }, 2]
]
```

So I use `minItems` to enforce that the the inner array has at least the name/path string item, as parameters are optional. The JSON schema allows any number of optional array items, unless there is a `additionalItems` parameter set to `false`, so the optional/repeating case is covered by default.

The overall goal here is to provide as much upfront validation as possible while editing these files. In the case of `minItems` and `minLength`, it looks like `markdownlint-cli2` would throw an error if [given an empty string](https://github.com/DavidAnson/markdownlint-cli2/blob/main/markdownlint-cli2.js#L49) for a `require` ID or if [given a zero-length array](https://github.com/DavidAnson/markdownlint-cli2/blob/main/markdownlint-cli2.js#L60) for the options that support parameters.

An added bonus of separating out the item types is tooltip support in the editor for the inner item descriptions of the arrays.

References:
[Arrays - JSON Schema](https://cswr.github.io/JsonSchema/spec/arrays/)
[Multiple Types - JSON Schema](https://cswr.github.io/JsonSchema/spec/multiple_types/)

Reply:
Now that I see it listed out, I think I like it better without type like you started with. But if "An added bonus" means this has better tool tips, lets stick with it. Sorry!